tracing-modality/src/common/attr_handlers.rs

```diff
@@ -1,15 +1,81 @@
+//! by `tracing`.
+//!
+//! ## Default Handlers
+//!
+//! By default, the following handlers are defined:
+//!
+//! | tracing key                      | modality key                           | value conversion         |
+//! | :---                             | :---                                   | :---                     |
+//! | "timestamp"                      | "event.timestamp"                      | see [timestamp]          |
+//! | "interaction.remote_timestamp"   | "event.interaction.remote_timestamp"   | see [remote_timestamp]   |
+//! | "interaction.remote_timeline_id" | "event.interaction.remote_timeline_id" | see [remote_timeline_id] |
+//! | "name"                           | "event.name"                           | see [name]               |
+//! | "message"                        | "event.message"                        | see [name]               |
+//! | "severity"                       | "event.severity"                       | see [severity]           |
+//! | "source.module"                  | "event.source.module"                  | see [source_module]      |
+//! | "source.file"                    | "event.source.file"                    | see [source_file]        |
+//!
+//! These default handlers can be overridden using the [Options::set_attr_handlers()](crate::Options::set_attr_handlers)
+//! or [Options::with_attr_handlers()](crate::Options::with_attr_handlers) functions.
+//!
+//! ## Fallback handler
+//!
+//! Additionally, there is a "fallback handler" that will take any tracing key NOT matched by the attribute
+//! handlers by ensuring that the given key starts with "event.*", and the contained data is converted into
+//! a format usable by modality. If this conversion is suitable for data logged by your application, then no
+//! specific attribute handler needs to be provided.
+//!
+//! At the moment, this fallback handler cannot be configured, replaced, or disabled.
+//!
+//! ## Tracing Metadata
+//!
+//! Additionally, `tracing-modality` will provide the following event keys with data sourced from `tracing`'s
+//! metadata fields, IF the values have not already been filled by any of the Default or Fallback handlers.
+//!
+//! | modality key             | tracing metadata (or other) source           |
+//! | :---                     | :---                                         |
+//! | "event.name"             | `metadata.name()`                            |
+//! | "event.severity"         | `metadata.level()`                           |
+//! | "event.source.module"    | `metadata.module_path()`                     |
+//! | "event.source.file"      | `metadata.file()`                            |
+//! | "event.source.line"      | `metadata.line()`                            |
+//! | "event.internal.rs.tick" | nanosecond ticks since first tracing event   |
+//! | "event.timestamp"        | nanosecond ticks since unix epoch            |
+//!
```

Comment:
This is _probably_ the wrong place for this. I think it might make more sense to just move this to the readme. Thoughts?